lib/response-handler.js

```diff
@@ -0,0 +1,168 @@
+"use strict";
+
+module.exports = (error, logger, parseLinkHeader) => {
+    /**
+     * @typedef {Object} artifactResponse
+     * @property {string} statusCode The HTTP status code.
+     * @property {string} message The status message for the response.
+     * @property {Object} body The response body.
+     * @property {string} location The url of the uploaded artifact.
+     */
+
+    /**
+     * Handles the response gotten from making requests to Shelf.
+     *
+     * @param {http~IncomingMessage} response
+     * @return {Promise.<Object>}
```

Comment:
Not necessary to return a promise here.  Any value will resolve. Promise chains are cool that way.